src/containers/Group/Providers/Base.js

```diff
@@ -0,0 +1,103 @@
+import React, { Component, PropTypes } from 'react';
+import { initialize } from 'redux-form';
+import Helmet from 'react-helmet';
+import { connect } from 'react-redux';
+import { asyncConnect } from 'redux-async-connect';
+
+import CreateProducerForm from 'components/forms/producers/Create';
+import { load as loadProviders } from 'redux/modules/providers/providers';
+import { create as createProducer } from 'redux/modules/producers/list';
+import { create as createSupplier } from 'redux/modules/suppliers/suppliers';
+import List from './List';
+
+const mapStateToProps = (state) => ({
+  createProducerErrors: state.producersReducer.createProducerErrors,
+  createSupplierErrors: state.suppliersReducer.createSupplierErrors,
+  providers: state.providersReducer.providers.entities,
+});
+
+const mapDispatchToProps = { initialize, createProducer, createSupplier };
+
+@asyncConnect([{
+  promise: (options) => {
+    const {
+      store: { dispatch },
+      params: { id },
+    } = options;
+
+    const promises = [];
+    promises.push(dispatch(loadProviders(id)));
+
+    return Promise.all(promises);
+  },
+}])
+@connect(mapStateToProps, mapDispatchToProps)
+export default class GroupSuppliersBase extends Component {
+  static propTypes = {
+    group: PropTypes.object.isRequired,
+    currentUser: PropTypes.object.isRequired,
+    providers: PropTypes.array.isRequired,
+    createProducer: PropTypes.func.isRequired,
+    createSupplier: PropTypes.func.isRequired,
+    createProducerErrors: PropTypes.object,
+    createSupplierErrors: PropTypes.object,
+  }
+
+  handleSubmit(data) {
+    const self = this;
+
+    data.group_id = this.props.group.id;
+
+    return this.props.createProducer(data).then(() => {
```

Comment:
What happen if you put `data` in this `then`? Do you see the pattern? :trollface: